constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java

```diff
@@ -53,22 +56,78 @@
     /**
      * the HTTP client used by all instances (with retry, only for idempotent requests like GET)
      */
-    private static OkHttpClient clientWithRetry = new OkHttpClient.Builder()
-        .addInterceptor(new ConstructorInterceptor())
+    private static OkHttpClient clientWithRetry = client.newBuilder()
         .retryOnConnectionFailure(true)
         .build();
 
     /**
      * @param newClient the HTTP client to use by all instances
      */
-    protected static void setClient(OkHttpClient newClient) {
-        client = newClient;
+    public static void setClient(OkHttpClient newClient) {
```

Comment:
Can we change this to `setOkHttpClient` and update the different places where it's used?

We should also update the param to specify OkHttpClient (i.e. `newOkHttpClient the OkHttpClient to use by all instances` or something) 🤔